### PR TITLE
Split utils

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**

### CODE REVIEW

File: src/client/visualizations/line-chart/charts/charts-per-split/label.tsx [hunk 1]
```diff
@@ -29,8 +29,9 @@ interface LabelProps {
 export const Label: React.SFC<LabelProps> = props => {
   const { essence, datum } = props;
   if (hasNominalSplit(essence)) {
+    const nominalSplit = getNominalSplit(essence);
     const nominalDimension = getNominalDimension(essence);
-    const splitValue = datum[nominalDimension.name] as PlywoodValue;
+    const splitValue = nominalSplit.selectValue(datum);
     return <div className="split-chart-label">
     <span className="split-chart-dimension-title">
       {nominalDimension.title}
```
Comment (kzadurska) [inline src/client/visualizations/line-chart/charts/charts-per-split/label.tsx:37]: Can we get title from split or do we need to get it from dimension?

File: src/client/visualizations/bar-chart/improved-bar-chart/bars/bars.tsx [hunk 2]
```diff
@@ -48,21 +45,20 @@ const Bar: React.SFC<BarProps> = props => {
       : <StackedBar {...rest} model={model} />;
   }
   return showPrevious
-    ? <SingleTimeShiftBar {...rest} />
-    : <SingleBar {...rest} />;
+    ? <SingleTimeShiftBar {...rest} model={model} />
+    : <SingleBar {...rest} model={model} />;
```
Comment (kzadurska) [inline src/client/visualizations/bar-chart/improved-bar-chart/bars/bars.tsx:49]: Would there be a difference if we did like this?
```suggestion
    : <SingleBar {...props} />;
```

Currently `{...rest} model={model}` makes me think there is some important operation done on `model` between taking it from props and passing it down to components 🤔

File: src/client/visualizations/line-chart/charts/charts-per-split/label.tsx (hunk 1 above)
Comment (adrianmroz-allegro) [reply src/client/visualizations/line-chart/charts/charts-per-split/label.tsx:37]: `title` is defined on Dimension (by user via configuration file). Dimensions are basically read-only but complex so we don't want to put them on splits which are often serialised and deserialised.

File: src/client/visualizations/bar-chart/improved-bar-chart/bars/bars.tsx (hunk 2 above)
Comment (adrianmroz-allegro) [reply src/client/visualizations/bar-chart/improved-bar-chart/bars/bars.tsx:49]: In your example, TS can't infer narrower type of `model`. We must be explicit for TS. Of course we could stitch `rest` and `model` together again with spread, but surprisingly correct TSX syntax is `<StackedTimeShiftBar {... { ...rest, model }}  />`

**Review by plesiecki — APPROVED** [review]

**Review by mkuthan — APPROVED** [review]
Would you mind to add some tests for Split.selectValue and Split.formatValue if feasible?

**Review by mkuthan — APPROVED** [review]

**Review by plesiecki — APPROVED** [review]

**Review by kzadurska — APPROVED** [review]
🚀

**Pull request merged**